vortex-array/src/compute/binary_search.rs

```diff
@@ -0,0 +1,34 @@
+use vortex_error::{vortex_err, VortexResult};
+
+use crate::array::{Array, WithArrayCompute};
+use crate::compute::search_sorted::{SearchSorted, SearchSortedSide};
+use crate::scalar::Scalar;
+
+pub trait BinarySearchFn {
+    fn binary_search(&self, value: &Scalar) -> VortexResult<Result<usize, usize>>;
+}
+
+pub fn binary_search<T: Into<Scalar>>(
```

Comment:
I'm not sure you need this actually? I think making SearchSorted return Result<usize, usize> is sufficient, and actually strictly more powerful since it lets you grab the right-hand-side of a run of values.

Reply:
I can agree with that